Relocate contributing guidelines (#2166)
meilisearch/documentation: learn/contributing/contributing_to_docs.md

If this happens to you and you think there has been some mistake, please let us know and we will try to rectify the situation.

## Local development

### Requirements

- [Node version](https://nodejs.org/en/) >= v14 and <= v16

### Installing and running the docs

```bash
# Clone the repository
git clone git@github.com:meilisearch/documentation.git meilisearch-documentation

# Open the newly created directory
cd meilisearch-documentation

# Install dependencies
yarn install

# Run Meilisearch documentation on http://localhost:8080
yarn dev
```

### Testing

A complete test can be done using the following command:

```bash
yarn test
```

The tests are triggered on build and on any pull request to main.

#### Checking dead links

Check for broken or dead links before submitting a pull request using:

```bash
yarn check-links
```

#### Checking styling

Run the following command to check for any styling errors like extra spaces or lines in `*.vue`, `*.js`, and `*.md` files:

> ```bash

Reviewer: GitHub is messing up the formatting when I try suggesting changes. Can you please fix the formatting for lines 147-149. It should be one code block wrapped in ```

yarn style

### Handling images and other assets

Screenshots, images, GIFs, and video demonstrations should be placed in a relevant folder under `.vuepress/public/` and then referenced from any markdown file a using relative link. For example, if you create the file `.vuepress/public/my_cool_guide/my_cool_image.png`, you would link to it from your document using 

> ```md

Reviewer: ```suggestion
```
```

If this happens to you and you think there has been some mistake, please let us know and we will try to rectify the situation.

## Local development

### Requirements

- Node version >= v14 and <= v16

### Installing & Running

```bash
# Clone the repository
git clone git@github.com:meilisearch/documentation.git meilisearch-documentation
# Open the newly created directory
cd meilisearch-documentation
# Install dependencies
yarn install
# Run Meilisearch documentation on http://localhost:8080
yarn dev
```

### Testing

A complete test can be done using the following command:

```bash
yarn test
```

The tests are triggered on build and on any pull request to main.

#### Checking dead links

Making changes to the documentation without creating any dead links can be a tedious task. Before submitting any pull request, you can check the dead links.

```bash
yarn check-links
```

#### Checking Styling

> The documentation follows styling rules. The following files will be checked: `*.vue`, `*.js`, and `*.md`.
> 
> You can check out and fix the styling errors.
> 
> ```bash
> yarn style:fix
> ```
> 
> You can test if the code is well-formatted without fixing it.
> 
> ```bash
> yarn style
> ```

Reviewer: ```suggestion
Run the following command to check for any styling errors like extra spaces or lines in `*.vue`, `*.js`, and `*.md` files:

```bash
yarn style
```

You can fix the styling errors using:

```bash
yarn style:fix
```
```

### Handling images and other assets

> Screenshots, images, GIFs, and video demonstrations should be placed in a relevant folder under `.vuepress/public/` and then referenced from any markdown file a using relative link. For example, if you create the file `.vuepress/public/my_cool_guide/my_cool_image.png`, you would link to it from your document using 

Reviewer: ```suggestion
Screenshots, images, GIFs, and video demonstrations should be placed in a relevant folder under `.vuepress/public/` and then referenced from any markdown file a using relative link. For example, if you create the file `.vuepress/public/my_cool_guide/my_cool_image.png`, you would link to it to your document using:
```

Author: Possibly a better wording would be "you would **embed it** in your document using:", neither of these two phrasings above sound particularly good to me. Thoughts?
